### PR TITLE
Scene import implementation

### PR DESCRIPTION
**Pull request by jturner65**
## Motivation and Context
This is the final functionality PR for the scene import functionality.  This PR supports json files holding scene-specific configurations to override the old scene filename-based functionality settings.  If json configuration files for a particular scene do not exist, the current mechanism for setting values based on file name/extension are still followed.

This PR will be followed by a tutorial PR on how to consume this new functionality.

## How Has This Been Tested
C++ tests needed to be modified to consume the new sceneAttributes-based protocol.  All existing c++ and python tests pass.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/esp/assets/ResourceManager.cpp [hunk 1]
```diff
@@ -167,98 +167,238 @@ void ResourceManager::initPhysicsManager(
...
+          << "ResourceManager::loadScene : Not loading semantic mesh - "
+             "File Name : "
+          << houseFile << " does not exist.";
+      activeSemanticSceneID = activeSceneIDs[0];
+    }
+  } else {  // not wanting to create semantic mesh
+    LOG(INFO) << "ResourceManager::loadScene : Not loading semantic mesh";
+  }
+  // save active semantic scene ID so that simulator can consume
+  activeSceneIDs[1] = activeSemanticSceneID;
```
Comment (jturner65) [inline src/esp/assets/ResourceManager.cpp:271]: Note the logic in the above block - some assumptions were made for edge cases that the original code did not cover.  Please review.

File: src/esp/assets/managers/SceneAttributesManager.cpp [hunk 2]
```diff
@@ -268,11 +371,70 @@ SceneAttributesManager::createFileBasedAttributesTemplate(
...
+  if (houseFName != "") {
+    // if "house filename" is specified in scene json to non-empty value, set
+    // value (override default).
+    sceneAttributes->setHouseFilename(houseFName);
+  }
+
+  if (io::jsonIntoVal<std::string>(jsonConfig, "lighting setup", lightSetup)) {
+    // if lighting is specified in scene json to non-empty value, set value
+    // (override default).
+    if (lightSetup != "") {
```
Comment (aclegg3) [inline src/esp/assets/managers/SceneAttributesManager.cpp:412]: Remove this if: "" is the default light key DEFAULT_LIGHTING_KEY, so we should accept this.

File: src/esp/assets/managers/SceneAttributesManager.cpp [hunk 3]
```diff
@@ -268,11 +371,70 @@ SceneAttributesManager::createFileBasedAttributesTemplate(
...
+      sceneAttributes->setLightSetup(lightSetup);
+    } else {
+      // if lighting is specified in scene json to empty value, set value to
+      // NO_LIGHT_KEY
+      sceneAttributes->setLightSetup(assets::ResourceManager::NO_LIGHT_KEY);
+    }
+    // // TODO: should potentially json-set "requires lighting" be overridden
+    // // existence of "lighting setup" specification?
+    // sceneAttributes->setRequiresLighting(
+    //     lightSetup != assets::ResourceManager::NO_LIGHT_KEY);
```
Comment (aclegg3) [inline src/esp/assets/managers/SceneAttributesManager.cpp:422]: It's fine as-is

File: src/esp/assets/managers/SceneAttributesManager.cpp [hunk 4]
```diff
@@ -286,6 +448,24 @@ SceneAttributesManager::createFileBasedAttributesTemplate(
...
+    io::JsonDocument& jsonDoc,
+    const char* jsonTag) {
+  std::string tmpVal = "";
+  io::jsonIntoVal<std::string>(jsonDoc, jsonTag, tmpVal);
+  if (tmpVal.length() != 0) {
+    std::string strToLookFor = Cr::Utility::String::lowercase(tmpVal);
+    if (AssetTypeNamesMap.count(tmpVal)) {
+      return static_cast<int>(AssetTypeNamesMap.at(tmpVal));
+    } else {
+      return static_cast<int>(AssetType::UNKNOWN);
```
Comment (aclegg3) [inline src/esp/assets/managers/SceneAttributesManager.cpp:461]: Provide a warning here for typos.

File: src/esp/assets/managers/SceneAttributesManager.cpp [hunk 5]
```diff
@@ -286,6 +448,24 @@ SceneAttributesManager::createFileBasedAttributesTemplate(
...
+    }
+  }
+  // -1 i
```
Comment (aclegg3) [inline src/esp/assets/managers/SceneAttributesManager.cpp:464]: remove

File: src/esp/assets/ResourceManager.cpp [hunk 6]
```diff
@@ -167,98 +167,238 @@ void ResourceManager::initPhysicsManager(
...
+        bool computeSemanticAABBs = splitSemanticMesh;
+        semanticSceneSuccess = loadSceneInternal(
+            semanticInfo, _physicsManager, &semanticRootNode,
+            &semanticDrawables, computeSemanticAABBs, splitSemanticMesh);
+        // regardless of load failure, original code still changed
+        // activeSemanticSceneID_
+        activeSceneIDs[1] = activeSemanticSceneID;
+        if (!semanticSceneSuccess) {
+          LOG(ERROR) << " ResourceManager::loadScene : Semantic scene mesh "
+                        "load failed.";
```
[the remark this answers is not in the document]
Comment (jturner65) [reply src/esp/assets/ResourceManager.cpp:249]: If any files in the semantic mesh load process fail to load, will return false with an error message.

**Pull request merged**